### ISSUE TITLE
Console commands ignore maintenance mode

### ISSUE THREAD

**Issue by podarcis** [pimcore/pimcore#1308]
The documentation for Console CLI states the console option `--ignore-maintenance-mode` (option gets added by Console\Application) is taken into account by Console\AbstractCommand. However, currently the check (more exact: throwing the exception) is disabled by commit 6000c07ea:
https://github.com/pimcore/pimcore/blob/4.4.3/pimcore/lib/Pimcore/Console/AbstractCommand.php#L63

Is this intended? If yes, maybe the option should be removed (or the uncommented code documented). If no, is re-enabling the exception safe regarding backward-compatibility?

**Comment by podarcis** [comment]
Here's the patch:
```patch
diff --git a/pimcore/lib/Pimcore/Console/AbstractCommand.php b/pimcore/lib/Pimcore/Console/AbstractCommand.php
index c95caf4..9612391 100644
--- a/pimcore/lib/Pimcore/Console/AbstractCommand.php
+++ b/pimcore/lib/Pimcore/Console/AbstractCommand.php
@@ -60,7 +60,7 @@ abstract class AbstractCommand extends \Symfony\Component\Console\Command\Comman
 
         // skip if maintenance mode is on and the flag is not set
         if (Admin::isInMaintenanceMode() && !$input->getOption('ignore-maintenance-mode')) {
-            //throw new \RuntimeException('In maintenance mode - set the flag --ignore-maintenance-mode to force execution!');
+            throw new \RuntimeException('In maintenance mode - set the flag --ignore-maintenance-mode to force execution!');
         }
     }

```

**Comment by cyawman** [comment]
Maybe this was commented out for a reason?

`php pimcore/cli/console.php update --source-build=4059 --update=4063 -vvv --ignore-maintenance-mode`

[2017-03-28 21:24:30] core.DEBUG: Executing command /usr/bin/php /data/www/pimcore/cli/console.php internal:update-processor {"type":"files","revision":4060} on the current shell {"origin":"Pimcore\\Tool\\Console::exec() on line 324","memory":"  9 MB"} []

  [RuntimeException]
  In maintenance mode - set the flag --ignore-maintenance-mode to force execution!

internal:update-processor [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [--ignore-maintenance-mode] [--maintenance-mode] [--environment [ENVIRONMENT]] [--] <command> [<config>]


 16/16 [============================] 100% 17 secs/17 secs 12.0 MiB

Update stopped by error! Please check your logs
Last return value was:

**Comment by cyawman** [comment]
This exception is being thrown even if I pass the --ignore-maintenance-mode flag from the CLI.

**Comment by brusch** [comment]
@cyawman I'm not able to reproduce it ...

**Comment by cyawman** [comment]
@brusch Anytime I try and upgrade with CLI from 4044 to 4063 I get an error at 4061 with a runtime exception about ignore maintenance mode. Even when I pass the flag it still errors out at 4061.

**Comment by podarcis** [comment]
It seems like the UpdateCommand calls `internal:update-processor` w/o the option `--ignore-maintenance-mode` (https://github.com/pimcore/pimcore/blob/4.5.0/pimcore/lib/Pimcore/Console/Command/UpdateCommand.php#L160), which is now (by this change) required while in maintenance mode? I'm afraid this broke the update process...

**Comment by cyawman** [comment]
Is there a CLI workaround for this issue?

**Comment by podarcis** [comment]
For the record: The regression is fixed by #1722 commit 690a130d7
